eclipse/sw360
# feat(ProjectUI): feature to display the source files linked with the licenses

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
* #1052

> Did you add or update any new dependencies that are required for your change?
* No

Issue: #1052 

### Suggest Reviewer
> You can suggest reviewers here with an @mention.

### How To Test?
> How should these changes be tested by the reviewer?
* Added a new column (`other licenses`) in `Tree View` table of `License Clearing` tab.
* Added an `info` icon beside each License (Main & Others), with tool tip `View file list`. (**screenshot 1**)
* Clicking on `info` icon should display the List of files using that License. (**screenshot 2**)
* A release should have only one `Approved CLI` in order to show the list of files, else appropriate error message will be shown. (**screenshot 3**)
* Added `Release Clearing State` filter (Issue #1102) in `List View` table of `License Clearing` tab (**screenshot 4**)
* Added `Expand All / Collapse All` feature (Issue #895) in `Tree View` table of `License Clearing` tab

### Screenshot 1:
![image](https://user-images.githubusercontent.com/50870174/116201606-4cf43c00-a757-11eb-9f2e-63bcc060e297.png)

### Screenshot 2:
![image](https://user-images.githubusercontent.com/50870174/116201516-32ba5e00-a757-11eb-981a-8617fcd26f2b.png)

### Screenshot 3:
![image](https://user-images.githubusercontent.com/50870174/116201925-a8bec500-a757-11eb-9d26-322b6838000c.png)

### Screenshot 4
![image](https://user-images.githubusercontent.com/50870174/116417729-2f0f0000-a859-11eb-94c7-adcaea0e3843.png)


Note: I have not modified the `List View` License Clearing table, to avoid the redundant information in both table.

> Have you implemented any additional tests?
* No

### Checklist
Must:

Signed-off-by: Abdul Kapti <abdul.kapti@siemens-healthineers.com>
